homeassistant/components/gocr/image_processing.py

```diff
@@ -0,0 +1,162 @@
+"""
+Local optical character recognition processing using GOCR.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing.gocr/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.core import split_entity_id
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, ImageProcessingEntity, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+
+import shlex
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_UNRECOGNIZED = 'unrecognized'
+CONF_CHARS = 'chars'
+CONF_EXTRA_ARGUMENTS = 'extra_arguments'
+CONF_HEIGHT = 'height'
+CONF_ROTATE = 'rotate'
+CONF_GOCR_BIN = 'gocr_bin'
+CONF_THRESHOLD = 'threshold'
+CONF_WIDTH = 'width'
+CONF_X_POSITION = 'x_position'
+CONF_Y_POSITION = 'y_position'
+CONF_NEGATE = 'negate'
+
+DEFAULT_BINARY = 'gocr'
+
+_KEY_GEOMETRY = 'geometry'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_EXTRA_ARGUMENTS, default=''): cv.string,
+    vol.Optional(CONF_UNRECOGNIZED): cv.string,
+    vol.Optional(CONF_CHARS): cv.string,
+    vol.Inclusive(CONF_HEIGHT, _KEY_GEOMETRY): cv.positive_int,
+    vol.Inclusive(CONF_X_POSITION, _KEY_GEOMETRY): cv.positive_int,
+    vol.Inclusive(CONF_WIDTH, _KEY_GEOMETRY): cv.positive_int,
+    vol.Inclusive(CONF_Y_POSITION, _KEY_GEOMETRY): cv.positive_int,
+    vol.Optional(CONF_GOCR_BIN, default=DEFAULT_BINARY): cv.string,
+    vol.Optional(CONF_THRESHOLD, default=50): cv.positive_int,
+    vol.Optional(CONF_ROTATE, default=0): cv.positive_int,
+
+    vol.Optional(CONF_NEGATE, default=False): cv.boolean,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the GOCR platform."""
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(ImageProcessingGocr(
+            hass, camera[CONF_ENTITY_ID], config, camera.get(CONF_NAME)
+        ))
+
+    async_add_entities(entities)
+
+
+class ImageProcessingGocr(ImageProcessingEntity):
+    """Representation of GOCR image processing entity."""
+
+    def __init__(self, hass, camera_entity, config, name):
+        """Initialize text processing."""
+        self._camera_entity = camera_entity
+        if name is not None:
+            self._name = name
+        else:
+            self._name = ("GOCR {0}".format(
+                split_entity_id(camera_entity)[1])
+                if config[CONF_NAME] == '' else
+                config[CONF_NAME])
+        self._state = None
+        if hasattr('config', CONF_X_POSITION) and config[CONF_X_POSITION] is not None:
```

Comment:
```suggestion
        if CONF_X_POSITION in config and config[CONF_X_POSITION] is not None:
```

Without that, the condition will always be false :)

Also, the check can be simplified to:

```suggestion
        if CONF_X_POSITION in config:
```